diff --git a/gradle.properties b/gradle.properties
index 39db198..ae45da8 100644
--- a/gradle.properties
+++ b/gradle.properties
@@ -1,2 +1,2 @@
-keycloakVersion=22.0.0
+keycloakVersion=23.0.4
 prometheusVersion=0.16.0
diff --git a/pom.xml b/pom.xml
index b84a44b..dcbd9a7 100644
--- a/pom.xml
+++ b/pom.xml
@@ -11,8 +11,9 @@

     <properties>
         <java.version>17</java.version>
-        <keycloak.version>22.0.0</keycloak.version>
+        <keycloak.version>23.0.4</keycloak.version>
         <prometheus.version>0.16.0</prometheus.version>
+        <quarkus-resteasy.version>3.2.9.Final</quarkus-resteasy.version>
         <version.compiler.maven.plugin>3.5.1</version.compiler.maven.plugin>
         <maven.compiler.source>17</maven.compiler.source>
         <maven.compiler.target>17</maven.compiler.target>
@@ -52,6 +53,12 @@
             <version>${keycloak.version}</version>
             <scope>provided</scope>
         </dependency>
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-resteasy-reactive</artifactId>
+            <version>${quarkus-resteasy.version}</version>
+            <scope>provided</scope>
+        </dependency>
         <dependency><!-- required by 'system-stubs-junit4' -->
             <groupId>org.mockito</groupId>
             <artifactId>mockito-inline</artifactId>
diff --git a/src/main/java/org/jboss/aerogear/keycloak/metrics/MetricsFilterProvider.java b/src/main/java/org/jboss/aerogear/keycloak/metrics/MetricsFilterProvider.java
index e013fa8..ddbe757 100644
--- a/src/main/java/org/jboss/aerogear/keycloak/metrics/MetricsFilterProvider.java
+++ b/src/main/java/org/jboss/aerogear/keycloak/metrics/MetricsFilterProvider.java
@@ -5,6 +5,8 @@ import jakarta.ws.rs.container.ContainerRequestFilter;
 import jakarta.ws.rs.container.ContainerResponseContext;
 import jakarta.ws.rs.container.ContainerResponseFilter;
 import jakarta.ws.rs.ext.Provider;
+import org.jboss.resteasy.reactive.server.ServerRequestFilter;
+import org.jboss.resteasy.reactive.server.ServerResponseFilter;

 /**
  * This provider registers the MetricsFilter within environments that use Resteasy 4.x and above, e.g. Keycloak.X.
@@ -13,11 +15,13 @@ import jakarta.ws.rs.ext.Provider;
 public class MetricsFilterProvider implements ContainerRequestFilter, ContainerResponseFilter {

     @Override
+    @ServerRequestFilter(preMatching = true)
     public void filter(ContainerRequestContext requestContext) {
         MetricsFilter.instance().filter(requestContext);
     }

     @Override
+    @ServerResponseFilter
     public void filter(ContainerRequestContext requestContext, ContainerResponseContext responseContext) {
         MetricsFilter.instance().filter(requestContext, responseContext);
     }
